lib/chef/provider/service/windows.rb

```diff
@@ -83,20 +83,12 @@ def load_current_resource
 
   def start_service
     if Win32::Service.exists?(@new_resource.service_name)
-      # reconfiguration is idempotent, so just do it.
-      new_config = {
-        service_name: @new_resource.service_name,
-        service_start_name: @new_resource.run_as_user,
-        password: @new_resource.run_as_password,
-      }.reject { |k, v| v.nil? || v.length == 0 }
-
-      Win32::Service.configure(new_config)
-      logger.info "#{@new_resource} configured."
+      action_configure
```

Comment:
To be clear, I don't think it makes sense to call configure here, but I left it so it doesn't break the existing API.